pkg/executables/config/kind.yaml

```diff
@@ -13,20 +13,20 @@ kubeadmConfigPatches:
         imageTag: {{.EtcdVersion}}
     imageRepository: {{.KubernetesRepository}}
     kubernetesVersion: {{.KubernetesVersion}}
-{{- if (ne .RegistryMirrorEndpoint "") }}
+{{- if .RegistryMirrorConfiguration }}
 containerdConfigPatches:
   - |
     [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
       [plugins."io.containerd.grpc.v1.cri".registry.mirrors."public.ecr.aws"]
-        endpoint = ["https://{{.RegistryMirrorEndpoint}}"]
-      [plugins."io.containerd.grpc.v1.cri".registry.configs."{{.RegistryMirrorEndpoint}}".tls]
+        endpoint = ["https://{{index .RegistryMirrorConfiguration "public.ecr.aws"}}"]
+      [plugins."io.containerd.grpc.v1.cri".registry.configs."{{index .RegistryMirrorConfiguration "base"}}".tls]
```

Comment:
This feels a little error prone and flaky because it pertains to a string defined in `constants` that itself living far from the core code.

We could create some functions we inject into the templater which would allow us to define the functions in Go code and clearly document them. Its similar to my other comments mentioning the use of `map[string]string` and how that's super difficult to appreciate semantics for as a maintainer.